addons/sale_management/views/sale_order_template_views.xml

```diff
@@ -48,7 +51,7 @@
                                 <field name="display_type" invisible="1"/>
                                 <group attrs="{'invisible': [('display_type', '!=', False)]}">
                                     <group>
-                                        <field name="product_id" attrs="{'required': [('display_type', '=', False)]}"/>
+                                        <field name="product_id" attrs="{'required': [('display_type', '=', False)]}" domain="[('sale_ok', '=', True)]"/>
```

Comment:
No need to change anything here, the domain of product_id is already defined in the python code.